cnd/src/network/comit_ln.rs

```diff
@@ -577,3 +577,281 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        ethereum::Address,
+        network::{derive_key_pair, transport, DialInformation, TokioExecutor},
+        swap_protocols::EthereumIdentity,
+    };
+    use anyhow::Context;
+    use bitcoin::secp256k1;
+    use futures::{pin_mut, task, StreamExt, TryFutureExt};
+    use libp2p::{
+        multihash::Sha3_256,
+        swarm::{SwarmBuilder, SwarmEvent},
+        Multiaddr, PeerId, Swarm,
+    };
+    use rand::thread_rng;
+
+    use crate::{asset::ethereum::FromWei, network::ComitNode};
+    use bitcoin::hashes::core::mem::swap;
+    use futures::{
+        future::{join3, join4},
+        join,
+        prelude::*,
+        ready,
+    };
+    use std::{str::FromStr, sync::Arc, thread::sleep, time};
+    use tokio::{macros::support::Pin, runtime, sync::Mutex};
+
+    fn random_swap_digest() -> SwapDigest {
+        SwapDigest::new(Sha3_256::digest(b"swapt digest"))
+    }
+
+    async fn start_communication(
+        alice_swarm: Arc<Mutex<Swarm<ComitLN>>>,
+        bob_swarm: Arc<Mutex<Swarm<ComitLN>>>,
+        alice_params: HanEtherereumHalightBitcoinCreateSwapParams,
+        bob_params: HanEtherereumHalightBitcoinCreateSwapParams,
+        alice_node_id: NodeLocalSwapId,
+        bob_node_id: NodeLocalSwapId,
+    ) {
+        println!("entered init...");
+
+        async {
+            // init bob first, because his swap has to be available for alice' announce
+            // message
+            sleep(time::Duration::from_secs(1));
+            let mut bob_guard = bob_swarm.lock().await;
+            bob_guard.initiate_communication(bob_node_id, bob_params);
+            println!("Bob initialised");
+
+            // init alice after bob
+            sleep(time::Duration::from_secs(1));
+            let mut alice_guard = alice_swarm.lock().await;
+            alice_guard.initiate_communication(alice_node_id, alice_params);
+            println!("Alice initialised");
+        }
+        .await;
+    }
+
+    #[test]
+    fn lightning_to_ethereum_integration_test() {
```

Comment:
nit: We currently only support Han/ether to HALight/bitcoin so this name seems a little misleading.